compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -890,6 +890,11 @@ impl<'tcx, 'a, Tag: Provenance, Extra> AllocRefMut<'a, 'tcx, Tag, Extra> {
     ) -> InterpResult<'tcx> {
         self.write_scalar(alloc_range(offset, self.tcx.data_layout().pointer_size), val)
     }
+
+    /// Mark the entire referenced range as uninitalized
+    pub fn write_uninit(&mut self) {
+        self.alloc.mark_init(self.range, false);
```

Comment:
Uh-oh, making `mark_init` public was a mistake. (Not sure when that happened, that was before this PR.)

This call is wrong, since it forgets to also clear the relocations in that range. Some other uses of `mark_init` are also wrong. I am working on a fix.